File: app/src/main/java/tv/spacedentist/android/view/SDButtonClickSender.java
package tv.spacedentist.android.view;

import android.util.Log;
import android.view.View;

import org.json.JSONException;

import tv.spacedentist.android.chromecast.SDChromecastManager;

public class SDButtonClickSender implements View.OnClickListener {

    private static final String TAG = "SDButtonClickSender";

    private final SDChromecastManager mChromecastManager;

    public SDButtonClickSender(SDChromecastManager chromecastManager) {
        mChromecastManager = chromecastManager;
    }

    @Override
    public void onClick(View v) {
        try {
            SDButton button = SDButton.fromResId(v.getId());
            Log.d(TAG, "Button was clicked: " + button.getKey());
            mChromecastManager.sendChromecastMessage(button.getMessage());
        } catch (JSONException e) {
            Log.e(TAG, "Exception while sending message", e);
        }
    }
}

File: app/src/main/java/tv/spacedentist/android/view/SDButtonClickSender.kt
package tv.spacedentist.android.view

import android.util.Log
import android.view.View

import org.json.JSONException

import tv.spacedentist.android.chromecast.SDCastManager

class SDButtonClickSender(private val mCastManager: SDCastManager) : View.OnClickListener {

    override fun onClick(v: View) {
        try {
            val button = SDButton.fromResId(v.id)
            Log.d(TAG, "Button was clicked: " + button.key)
            mCastManager.sendCastMessage(button.message)
        } catch (e: JSONException) {
            Log.e(TAG, "Exception while sending message", e)
        }

    }

    companion object {
        private const val TAG = "SDButtonClickSender"
    }
}
